backend/payment.php

```diff
@@ -1,29 +1,88 @@
 <?php
-require_once __DIR__.'/lib/Stripe.php';
-require_once __DIR__.'/config.loader.php';
+require_once __DIR__ . '/lib/Stripe.php';
+require_once __DIR__ . '/config.loader.php';
+require_once __DIR__ . '/secure.functions.php';
+require_once __DIR__ . '/stripe.cusid.functions.php';
 
 Stripe::setApiKey($config['stripe_sk']);
 
 if (isset($_POST['token'])) {
-    $token  = $_POST['token'];
-    $amount = $_POST['amount'];
-    $email = $_POST['email'];
-
-    // Create the charge on Stripe's servers - this will charge the user's card
-    try {
-        $charge = Stripe_Charge::create(array(
-            'amount' => $amount,
-            'currency' => 'usd',
-            'card' => $token,
-            'description' => 'elementary OS download',
-            'receipt_email' => $email,
-        ));
-        // Set an insecure, HTTP only cookie for 10 years in the future.
-        setcookie('has_paid_freya', $amount, time() + 315360000, '/', '.elementary.io', 0, 1);
-        echo 'OK';
-    } catch(Stripe_CardError $e) {
-        echo 'error';
-    }
+	$token = $_POST['token'];
+	$amount = $_POST['amount'];
+	$email = $_POST['email'];
+
+	// Create the charge on Stripe's servers - this will charge the user's card
+	try {
+
+		$stripecusid = customerHasStripeAccont($email);
+		$customer = null;
+
+		//check if the customer has stripe account with our db
+		if ($stripecusid === null) {
+			$customer = Stripe_Customer::create(array(
+				// "source" => $token ,
+				"email" => $email,
+				"source" => $token,
+			));
+			$stripecusid = $customer->id;
+		} else {
+
+			$customerdeltedatStripe = false;
```

Comment:
Yes sorry  $customerdeletedatstripe  sorry i am really suck in
typing/speaking in  English !

and i totally for got to add sample config and wrapping up the database
operations with crypto  functions i will push tomorrow

Cheers,

{
  "name": "*Charan Raj T C*",
  "email": "charanraj.tc@gmail.com <mailto:charanraj.tc@gmail.com>",
  "telephone": "+971 55 1238682 <+971%2055%201238682>",
  "url": "http://coresoft.co.in/charanraj.tc"
}                            	


On Sat, Feb 6, 2016 at 9:58 PM, Blake Kostner <notifications@github.com>
wrote:

> In backend/payment.php
> <https://github.com/elementary/mvp/pull/1013#discussion_r52104917>:
>
> > +	try {
> > +
> > +		$stripecusid = customerHasStripeAccont($email);
> > +		$customer = null;
> > +
> > +		//check if the customer has stripe account with our db
> > +		if ($stripecusid === null) {
> > +			$customer = Stripe_Customer::create(array(
> > +				// "source" => $token ,
> > +				"email" => $email,
> > +				"source" => $token,
> > +			));
> > +			$stripecusid = $customer->id;
> > +		} else {
> > +
> > +			$customerdeltedatStripe = false;
>
> Is $customerdeltedatStripe supposed to be $customerdeletedatstripe ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elementary/mvp/pull/1013/files#r52104917>.
>